### ISSUE TITLE
[Report] - INCORRECT information 

### ISSUE THREAD

**Issue by ngochd7** [golang/tour#1478]
Context: https://go.dev/tour/concurrency/4

I found the following content that you provided is not correct: "The loop for i := range c receives values from the channel repeatedly until it is closed"

It’s possible to close a non-empty channel but still have the remaining values be received. Here is example code:

```go
package main

import "fmt"

func main() {

	queue := make(chan string, 3)
	queue <- "one"
	queue <- "two"
	queue <- "three"
	close(queue)

	for elem := range queue {
		fmt.Println(elem)
	}
}
```

**Comment by ngochd7** [comment]
Yes, I agree with you, but I am raising `the problem of For loop with Channel`.

Different ways of using Channels solve different problems:
- Using the `msg, ok := <- ch` syntax will help you check if a channel is closed or not.
- Using `For` instead of `Select-Case` syntax will help you get all the msg in the Channel even if it is closed. When a channel is closed and there are no more msg in it, the For loop will terminate on its own.

**Comment by crisman** [comment]
I am no longer sure what you are reporting.  Could you rephrase the issue?

**Comment by ngochd7** [comment]
The problem is that you are not correctly expressing the end time of the For loop.
I think I detailed it in the first comment, please read it again carefully :333

**Comment by ngochd7** [comment]
I think you should separate them so others can see 2 things:
1. They can use the For syntax to get the messages out of the Channel
2. Using the For syntax will allow them to continue to retrieve messages that are still present in the Channel even after the Channel is closed.

It would be great if you have a detailed description of the differences of the 3 syntaxes:
- `msg := <- ch`
- `msg, ok := <- ch`
- `for msg := range ch`

**Comment by crisman** [comment]
For is not unique in allowing channel receive after a close.  Close does not remove items from a channel, nor does it stop anyone from receiving them.

To quote the slide:
> A sender can `close` a channel to indicate that no more values will be sent.

Those three options seem to me to be covered already.
